core/trino-main/src/main/java/io/trino/operator/OperatorStats.java

```diff
@@ -165,6 +167,7 @@ public OperatorStats(
         this.inputDataSize = requireNonNull(inputDataSize, "inputDataSize is null");
         checkArgument(inputPositions >= 0, "inputPositions is negative");
         this.inputPositions = inputPositions;
+        this.inputPositionsDistribution = requireNonNull(inputPositionsDistribution, "inputPositionsDistribution is null");
```

Comment:
We should consider using the generic operator metrics facility we added recently, instead of implementing yet another metric (and everything that's needed to handle merging values, reporting, etc)

Reply:
I guess this can be done, but I think we need to extend `io.trino.spi.metrics.Metric` to support dropping of intermediate structures (see https://github.com/trinodb/trino/pull/10124#discussion_r759603896)